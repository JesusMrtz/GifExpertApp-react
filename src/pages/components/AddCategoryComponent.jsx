import { useState } from "react";
import PropTypes from 'prop-types';


export function AddCategoryComponent({ onNewCategory }) {
    const [inputValue, setInputValue] = useState('');

    function onInputChanged({ target: { value } }) {
        setInputValue( value );
    }

    function onSubmit(e) { 
        e.preventDefault();
        const value = inputValue.trim();

        if ( !value.length ) return;
        
        onNewCategory(value);
        setInputValue('');
    }

    return(
        <>
            <form onSubmit={ onSubmit } role="form">
                <input type="text" placeholder="Buscar GIF's" value={ inputValue } onChange={ onInputChanged } />
            </form>
        </>
    );
}


AddCategoryComponent.propTypes = {
    onNewCategory: PropTypes.func.isRequired
}