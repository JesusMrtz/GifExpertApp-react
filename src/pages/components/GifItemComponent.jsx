import PropTypes from 'prop-types';


export function GifItemComponent({ image }) {
    return (
        <div className="card">
            <img src={ image.url } alt={ image.title } />
            <p>{ image.title }</p>
        </div>
    );
}


GifItemComponent.propTypes = {
    image: PropTypes.object.isRequired
}