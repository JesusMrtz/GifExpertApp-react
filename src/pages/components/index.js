export * from './AddCategoryComponent';
export * from './GifGridComponent';
export * from './GifItemComponent';