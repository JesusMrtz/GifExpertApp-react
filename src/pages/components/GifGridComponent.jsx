import PropTypes from 'prop-types';

import { userFetchGifs } from "../hooks/useFetchGifs";
import { GifItemComponent } from "./GifItemComponent";


export function GifGridComponent({ category }) {
    const { images, isLoading } = userFetchGifs(category);

    return(
        <>
            <h1>{ category }</h1>
            {
                isLoading && ( <h2>Cargando...</h2> )
            }
            
            <div className="card-grid">
                {
                    images.map((image) => (
                        // Aqui estoy esparciendo las properties para no escribir cada una que necesite el componente hijo
                        // <GifItemComponent key={ image.id } {...image} />

                        <GifItemComponent key={ image.id } image={ image } />
                    ))
                }
            </div>
        </>
    );
}


GifGridComponent.propTypes = {
    category: PropTypes.string.isRequired
}