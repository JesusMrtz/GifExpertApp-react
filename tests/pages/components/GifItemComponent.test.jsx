import { render, screen } from "@testing-library/react";
import { GifItemComponent } from '../../../src/pages/components/GifItemComponent';


describe(`Pruebas en el componente <GifItemComponent />`, () => {
    const imageMock = {
        id: '1',
        title: 'Spiderman',
        url: 'https://www.google.com/'
    }

    it('Debe de hacer match con el snaptshot', () => {
        // Renderiza un componente en memoria
        // actualiza el objeto "screen" que es parte del @testing-library
        const { container } = render(<GifItemComponent image={ imageMock } />);
        expect( container ).toMatchSnapshot();
    });


    it('Debe de mostrar la imagen con el SRC y el ALT', () => {
        render(<GifItemComponent image={ imageMock } />);
        const getImage = screen.getByRole('img');
        // screen.debug();

        expect( imageMock.url ).toBe( getImage.src );
        expect( imageMock.title ).toBe( getImage.alt );
    });
});