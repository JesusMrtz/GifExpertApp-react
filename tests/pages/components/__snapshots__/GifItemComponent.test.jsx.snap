// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <GifItemComponent /> Debe de hacer match con el snaptshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Spiderman"
      src="https://www.google.com/"
    />
    <p>
      Spiderman
    </p>
  </div>
</div>
`;
